### ISSUE TITLE
[not an issue] API-Killer-Beacon simple request out of topic

### ISSUE THREAD

**Issue by Decopi** [eladkarako/chrome_extensions#11]
Hi @eladkarako !

I saw today some of your add-ons being updated, and I'm retesting them. Please, two simple requests:

1) Please can you tell me if [this add-on WebAPIBlocker](https://addons.mozilla.org/en-US/firefox/addon/webapi-blocker/) already includes the beacon block function in equal way than your add-on API-Killer-Beacon blocks? Please, same question comparing your other API-Killer add-ons with this WebAPIBlocker add-on. Is there any special reason you choose to build separate add-ons instead of all together like WebAPIBlocker?

2) Out of topic from your add-on: Please, can you take a look at the API list inside [WebAPIBlocker add-on](https://addons.mozilla.org/en-US/firefox/addon/webapi-blocker/), and help me to understand from this list which APIs may be considered:
a) Relatively harmless to be blocked?
b) Medium?
c) Aggressive?

In simple words, if possible, I would like your help in order to define 3 blocking modes (LIGHT, CONSERVATIVE, AGGRESSIVE) based on the APIs list inside this add-on.
Yeah, this request is not related to your add-on, so of course, if you can't or don't want to help me, I perfectly understand, it will be fine. I asked the same request to the WebAPIBlocker' Dev, but sadly I haven't received answer from him. And you are the only Dev creating similar add-ons (API blockers). That's the reason I request help from you.

Thank you in advance!

**Comment by eladkarako** [comment]
checking (1).....

**Comment by eladkarako** [comment]
as far as I can see from this part:

```js
  "default": {
    "WebSocket": false,
    "MIDIInput": false,
    "SVGElement": false,
    "SVGAElement": false,
    "WebGLBuffer": false,
    "AudioContext": false,
    "HTMLCanvasElement": false
  }
```

it does not block beacon/PING requests.

still checking....
will improve my answer....

**Comment by eladkarako** [comment]
cool concept, 
in theory this web-extension "browse" all of the functions/objects creations from under 'window', 
but it seems to have a problem: it does do a good job regarding "finding all the APIs"
it just lists all the "functions" under the "window" object, 
but since not all of the HTML5 API functions lays under 'window'...
You can for example block "alert" function if you want.
but try blocking <code>window.navigator.sendBeacon</code> with this - <strong>you can't</strong>, 

code'wise - the null'ification method isn't working soon enough.

<img src="https://i.imgur.com/HdZrOcr.png" />

secondly I (actual code) use:

```javascript
"use strict";
var API = chrome || browser;

API.webRequest.onBeforeRequest.addListener(function(){return {cancel: true};}
, {urls:  ["<all_urls>"]
  ,types: ["ping"]
  }
, ["blocking"]
);
```
to block (<strong>just like a firewall</strong>) all the PING/BEACON requests that were generated by the page, so they will never leave your browser (won't connect to the remote server).

third, 
Google uses a new way of PINGing now, 
with a 'ping' attribute (see https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a )


to sum it up:
1. it does not handle blocking PING/BEACON at all since it does not find the functions recursively but just the ones directly 'under window'.
2. no firewall, if page generates PING it will pass through.
3. does not handle PING attribute (used by Google search page, implemented by browser).
4. null-ification process is too slow to start, in the page life cycle, and looks weak (I will not expand on this point).

**Comment by eladkarako** [comment]
I've removed that web-extension, without answering your '(2)'... it made me feel Yuckky!
as far as I could see there is no set of rules to rate any functions at all (risky/safe), 
b.t.w. this sounds like a very hard-labor, ones that does not pays off to put in a web-extension (similar to 'signatures' of anti-virus' - those require teams of specialists, and they too some times get things wrong, beside, no API is really "risky", it is all depend what you want to keep/kill...) 

That web-extension had some nice UI though (I'll give it a B+)
actual blocking is weak (I'll give it a hard C-)
